sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/booleanExpressions.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
+import org.apache.spark.sql.types.BooleanType
+
+/**
+ * String to indicate which boolean test selected.
+ */
+object BooleanTest {
+  val TRUE = "TRUE"
+  val FALSE = "FALSE"
+  val UNKNOWN = "UNKNOWN"
+
+  def calculate(input: Any, booleanValue: String): Boolean = {
+    booleanValue match {
+      case TRUE => input == true
+      case FALSE => input == false
+      case UNKNOWN => input == null
+      case _ => throw new AnalysisException("Boolean test value must be one of TRUE, " +
+        "FALSE and UNKNOWN.")
+    }
+  }
+}
+
+/**
+ * Test the value of an expression is true, false, or unknown.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(expr, booleanValue) - Returns true if `expr` equals booleanValue, " +
+    "or false otherwise.",
+  arguments = """
+    Arguments:
+      * expr - a boolean expression
+      * booleanValue - a boolean value represented by a string. booleanValue must be one
+          of TRUE, FALSE and UNKNOWN.
+  """,
+  examples = """
+    Examples:
+    > SELECT _FUNC_(1 > 2, true);
+       false
+    > SELECT _FUNC_(2 > 1, true);
+       true
+  """)
+case class BooleanTest(child: Expression, booleanValue: String)
```

Comment:
As another design choice, how about handling`IS UNKNOWN` just as `IS NULL`, then changing `booleanValue: String` into `booleanValue: Boolean`?

Reply:
There exists some different between the two syntax.
https://www.postgresql.org/docs/9.1/functions-comparison.html
have a lot description:
`Notice that IS UNKNOWN and IS NOT UNKNOWN are effectively the same as IS NULL and IS NOT NULL, respectively, except that the input expression must be of Boolean type.`

Reply:
Yea, I know the difference. So, I meant we cannot handle the difference in the `IsNull` side? (Just a question and I'm not sure the this is the best approach...)

Reply:
OK. thanks for your reminder. I handle the difference in `checkBooleanTestArgs`.
```
def checkBooleanTestArgs(e: Expression): Unit = e.dataType match {
    case BooleanType | NullType =>
    case other => throw new ParseException("argument of boolean test must be boolean or null, " +
	        s"not type $other", ctx)
}
```

Reply:
@dongjoon-hyun coud you advise us on this?

Reply:
@maropu @dongjoon-hyun Please take a look at this PR. Thanks.

Reply:
Oops. Sorry, @maropu and @beliefer . I'll try to review this PR tomorrow.

Reply:
+1 for @maropu 's direction. Let's reuse the existing one.

Reply:
OK. I will change it.